tests/test_commands.py

```diff
@@ -292,3 +304,192 @@ def test_monitor_usage_warning_sent(client, cli_runner, capfd):
+        "Units using DDS in production",
+        "Researchers",
+        "Unit users",
+        "Total number of users",
+    ]
+    first_row: typing.List = [f"2022-12-10", 2, 108, 11, 119]
+
+    # Fill reporting file with headers and one row
+    with reporting_file.open(mode="a") as csv_file:
+        writer = csv.writer(csv_file)
+        writer.writerow(header)  # Header - Columns
+        writer.writerow(first_row)  # First row
+
+    time_now = datetime(year=2022, month=12, day=10, hour=10, minute=54, second=10)
+    with freezegun.freeze_time(time_now):
+        # Run scheduled job now
+        with mock.patch.object(flask_mail.Mail, "send") as mock_mail_send:
+            cli_runner.invoke(reporting_units_and_users)
+            assert mock_mail_send.call_count == 1
+
+    # Check correct numbers
+    num_units: int = models.Unit.query.count()
+    num_users_total: int = models.User.query.count()
+    num_unit_users: int = models.UnitUser.query.count()
+    num_researchers: int = models.ResearchUser.query.count()
+    num_superadmins: int = models.SuperAdmin.query.count()
+    num_users_excl_superadmins: int = num_users_total - num_superadmins
+
+    # Expected new row:
+    new_row: typing.List = [
+        f"{time_now.year}-{time_now.month}-{time_now.day}",
+        num_units,
+        num_researchers,
+        num_unit_users,
+        num_users_excl_superadmins,
+    ]
+
+    # Check csv file contents
+    with reporting_file.open(mode="r") as result:
+        reader = csv.reader(result)
+        line: int = 0
+        for row in reader:
+            if line == 0:
+                assert row == header
+            elif line == 1:
+                assert row == [str(x) for x in first_row]
+            elif line == 2:
+                assert row == [str(x) for x in new_row]
+            line += 1
+
+    # Delete file to test error
+    fs.remove(reporting_file)
+    assert not fs.exists(reporting_file)
+
+    # Test no file found
+    with freezegun.freeze_time(time_now):
+        # Run scheduled job now
+        with mock.patch.object(flask_mail.Mail, "send") as mock_mail_send:
+            with pytest.raises(Exception) as err:
+                cli_runner.invoke(reporting_units_and_users)
+                assert mock_mail_send.call_count == 1
+            assert str(err.value) == "Could not find the csv file."
+
+    # Change total number of users to test error
+    with mock.patch("dds_web.scheduled_tasks.sum") as mocker:
+        mocker.return_value = num_users_total + 1
+        # Test incorrect number of users
+        with freezegun.freeze_time(time_now):
+            # Run scheduled job now
+            with mock.patch.object(flask_mail.Mail, "send") as mock_mail_send:
+                with pytest.raises(Exception) as err:
+                    cli_runner.invoke(reporting_units_and_users)
+                    assert mock_mail_send.call_count == 1
+                assert str(err.value) == "Sum of number of users incorrect."
```

Comment:
Note the `scheduled_tasks` -> `commands` as well as the exceptions
```suggestion
    # Change total number of users to test error
    with mock.patch("dds_web.commands.sum") as mocker:
        mocker.return_value = num_users_total + 1
        # Test incorrect number of users
        with freezegun.freeze_time(time_now):
            # Run scheduled job now
            with mock.patch.object(flask_mail.Mail, "send") as mock_mail_send:
                result: click.testing.Result = cli_runner.invoke(reporting_units_and_users)
                assert str(result.exception) == "Sum of number of users incorrect."
                assert mock_mail_send.call_count == 1
```